File: app/src/main/java/com/jann_luellmann/thekenapp/data/model/Customer.java
package com.jann_luellmann.thekenapp.data.model;

import com.jann_luellmann.thekenapp.R;
import com.jann_luellmann.thekenapp.data.util.Editable;

import androidx.annotation.NonNull;
import androidx.room.ColumnInfo;
import androidx.room.Entity;
import androidx.room.PrimaryKey;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Entity
public class Customer implements Comparable<Customer>{

    @PrimaryKey(autoGenerate = true)
    private long customerId;

    @ColumnInfo(name = "name")
    @Editable(stringId = R.string.name)
    private String name;

    @NonNull
    @Override
    public String toString() {
        return name;
    }

    @Override
    public int compareTo(Customer c) {
        return this.name.compareTo(c.getName());
    }
}

File: app/src/main/java/com/jann_luellmann/thekenapp/data/model/Customer.kt
package com.jann_luellmann.thekenapp.data.model

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.PrimaryKey
import com.jann_luellmann.thekenapp.R
import com.jann_luellmann.thekenapp.data.util.Editable

@Entity
data class Customer(
    @PrimaryKey(autoGenerate = true)
    var customerId: Long = 0,

    @ColumnInfo(name = "cname")
    @field:Editable(stringId = R.string.name)
    var name: String = ""
) : Comparable<Customer> {

    override fun toString(): String {
        return name
    }

    override fun compareTo(other: Customer): Int {
        return name.compareTo(other.name)
    }
}